scripts/tests/test_zkai_d128_component_compact_preprocessed_reprove_gate.py

```
@@ -0,0 +1,142 @@
+        self.assertIn("not a matched NANOZK benchmark", payload["non_claims"])
+        self.assertEqual(payload["mutations_checked"], len(gate.MUTATION_NAMES))
+        self.assertEqual(payload["mutations_rejected"], len(gate.MUTATION_NAMES))
+        self.assertTrue(payload["all_mutations_rejected"])
+
+    def test_individual_mutations_reject(self):
+        summary = cli_summary()
+        budget = prior_budget()
+        payload = gate.build_payload(summary, budget, include_mutations=False)
+        for name in gate.MUTATION_NAMES:
+            mutated = gate.mutate_payload(payload, name)
+            with self.assertRaises(gate.CompactPreprocessedGateError, msg=name):
```

Ruff (0.15.12)

[warning] Use `pytest.raises` instead of unittest-style `assertRaises`

(PT027)